crates/router/src/core/fraud_check.rs

```diff
@@ -433,85 +446,108 @@ pub async fn pre_payment_frm_core<'a, F>(
+                if matches!(frm_fraud_check.frm_status, FraudCheckStatus::Fraud) {
                     *should_continue_transaction = false;
-                    frm_info.suggested_action = Some(FrmSuggestion::FrmCancelTransaction);
-                } else if matches!(frm_configs.frm_action, api_enums::FrmAction::ManualReview) {
-                    *should_continue_capture = false;
-                    frm_info.suggested_action = Some(FrmSuggestion::FrmManualReview);
+                    if matches!(frm_configs.frm_action, api_enums::FrmAction::CancelTxn) {
+                        frm_info.suggested_action = Some(FrmSuggestion::FrmCancelTransaction);
+                    } else if matches!(frm_configs.frm_action, api_enums::FrmAction::ManualReview) {
+                        *should_continue_capture = false;
+                        frm_info.suggested_action = Some(FrmSuggestion::FrmManualReview);
+                    }
                 }
+                logger::debug!(
+                    "frm_updated_data: {:?} {:?}",
+                    frm_info.fraud_check_operation,
+                    frm_info.suggested_action
+                );
+                Some(frm_data_updated)
+            } else if matches!(
+                frm_configs.frm_preferred_flow_type,
+                api_enums::FrmPreferredFlowTypes::Post
+            ) {
+                *should_continue_capture = false;
+                Some(frm_data.to_owned())
+            } else {
+                Some(frm_data.to_owned())
             }
-            logger::debug!(
-                "frm_updated_data: {:?} {:?}",
-                frm_info.fraud_check_operation,
-                frm_info.suggested_action
-            );
-            Ok(Some(frm_data_updated))
         } else {
-            Ok(Some(frm_data.to_owned()))
-        }
-    } else {
-        Ok(None)
+            None
+        };
     }
+    Ok(frm_data)
 }
 
 #[allow(clippy::too_many_arguments)]
 pub async fn post_payment_frm_core<'a, F>(
     state: &AppState,
+    req_state: ReqState,
     merchant_account: &domain::MerchantAccount,
     payment_data: &mut payments::PaymentData<F>,
     frm_info: &mut FrmInfo<F>,
     frm_configs: FrmConfigsObject,
     customer: &Option<domain::Customer>,
     key_store: domain::MerchantKeyStore,
+    should_continue_capture: &mut bool,
 ) -> RouterResult<Option<FrmData>>
 where
     F: Send + Clone,
 {
     if let Some(frm_data) = &mut frm_info.frm_data {
-        // Allow the Post flow only if the payment is succeeded,
+        // Allow the Post flow only if the payment is authorized,
         // this logic has to be removed if we are going to call /sale or /transaction after failed transaction
         let fraud_check_operation = &mut frm_info.fraud_check_operation;
-        if payment_data.payment_attempt.status == AttemptStatus::Charged {
+        if payment_data.payment_attempt.status == AttemptStatus::Authorized {
```

Comment:
What if the manual_review is not configured, the fraud_check will only check for Authorized status?

Reply:
Yes, We have updated capture method to manual for all post frm flows. This will make payment always to authorized. If it is configured to auto capture by merchant. We store that in payment_capture_method in fraud_check table, then if it found to legit -> capture , fraud -> fraud_action(cancel or hold).